### PR TITLE
chore(cd): add all locks' message, id, author, timestamp to the progosis endpoints

### PR DESCRIPTION
**Pull request by AminSlk**
Ref: SRX-7CBX2O

### CODE REVIEW

File: services/cd-service/pkg/repository/transformer.go [hunk 1]
```diff
@@ -3787,44 +3800,50 @@ func (c *envReleaseTrain) prognosis(
...
 			if len(teamLocks) > 0 {
-				// we don't really care about multiple locks, since they all have the same effect, so we just pick one:
-				var firstLock Lock
-				sortedKeys := sorting.SortKeys(teamLocks)
-				for keyIndex := range sortedKeys {
-					key := sortedKeys[keyIndex]
-					firstLock = teamLocks[key]
-					break
+				locksList := []*api.Lock{}
+				sortedKeys := sorting.SortKeys(appLocks)
```
Comment (miguel-crespo-fdc) [inline services/cd-service/pkg/repository/transformer.go:3812]:
```suggestion
				sortedKeys := sorting.SortKeys(teamLocks)
```

File: services/cd-service/pkg/repository/transformer.go [hunk 2]
```diff
@@ -3787,44 +3800,50 @@ func (c *envReleaseTrain) prognosis(
...
+				for _, lockId := range sortedKeys {
+					newLock := &api.Lock{
+						Message:   envLocks[lockId].Message,
+						CreatedAt: timestamppb.New(envLocks[lockId].CreatedAt),
```
Comment (miguel-crespo-fdc) [inline services/cd-service/pkg/repository/transformer.go:3816]: this should be team locks 
```suggestion
						CreatedAt: timestamppb.New(team[lockId].CreatedAt),
```

File: services/cd-service/pkg/repository/transformer.go [hunk 3]
```diff
@@ -3690,29 +3697,35 @@ func (c *envReleaseTrain) prognosis(
...
+			locksList := []*api.Lock{}
 			sortedKeys := sorting.SortKeys(appLocks)
-			for keyIndex := range sortedKeys {
-				key := sortedKeys[keyIndex]
-				firstLock = appLocks[key]
-				break
+			for _, lockId := range sortedKeys {
+				newLock := &api.Lock{
+					Message:   envLocks[lockId].Message,
+					CreatedAt: timestamppb.New(envLocks[lockId].CreatedAt),
```
Comment (miguel-crespo-fdc) [inline services/cd-service/pkg/repository/transformer.go:3713]:
```suggestion
					CreatedAt: timestamppb.New(appLocks[lockId].CreatedAt),
```

**Review by miguel-crespo-fdc — CHANGES_REQUESTED** [review]

File: services/cd-service/pkg/repository/transformer.go (hunk 1 above)
Comment (AminSlk) [reply services/cd-service/pkg/repository/transformer.go]: Done

File: services/cd-service/pkg/repository/transformer.go (hunk 3 above)
Comment (AminSlk) [reply services/cd-service/pkg/repository/transformer.go]: Done

File: services/cd-service/pkg/repository/transformer.go (hunk 2 above)
Comment (AminSlk) [reply services/cd-service/pkg/repository/transformer.go]: Done

**Review by miguel-crespo-fdc — APPROVED** [review]

**Pull request merged**